src/main/java/nu/studer/teamcity/buildscan/internal/Util.java

```diff
@@ -6,7 +6,11 @@
 
 final class Util {
 
-    private static final Pattern BUILD_SCAN_URL_PATTERN = Pattern.compile("https?://.*/s/(.*)");
+    // Includes:
+    // 1. Optional `[INFO] ` prefix, which is present in Maven scans
+    // 2. A capturing group for the full URL (necessary for Maven scans)
+    // 3. A capturing group for the scan ID (necessary for the `BuildScanReference` instance).
+    private static final Pattern BUILD_SCAN_URL_PATTERN = Pattern.compile("(?:\\[INFO] )?(https?://.*/s/(.*))");
```

Comment:
@autonomousapps what is the `?:` for in this reg exp?

Reply:
`?:`, when placed at the start of a group, indicates a non-capturing group. I can index the other two groups as if the first wasn't there.

Reply:
Ah cool. Thanks for the quick and insightful response.